core/src/main/java/org/elasticsearch/script/ScriptContextRegistry.java

```diff
@@ -76,15 +81,11 @@ private void validateScriptContext(ScriptContext.Plugin scriptContext) {
         }
     }
 
-    private static ImmutableSet<String> reservedScriptContexts() {
-        ImmutableSet.Builder<String> builder = ImmutableSet.builder();
-        for (ScriptService.ScriptType scriptType : ScriptService.ScriptType.values()) {
-            builder.add(scriptType.toString());
-        }
-        for (ScriptContext.Standard scriptContext : ScriptContext.Standard.values()) {
-            builder.add(scriptContext.getKey());
-        }
-        builder.add("script").add("engine");
-        return builder.build();
+    private static Set<String> reservedScriptContexts() {
+        Set<String> reserved = concat(stream(ScriptService.ScriptType.values()), stream(ScriptContext.Standard.values()))
+                .map(Object::toString).collect(toSet());
```

Comment:
hmm, we need to call `getKey`, not `toString` on the elements of the second array?